variables.tf

```diff
@@ -7,6 +7,11 @@ variable "location" {
 variable "name" {
   type        = string
   description = "The name of the this resource."
+
+  validation {
+    condition     = can(regex("^[0-9A-Za-z-]+$", var.name))
+    error_message = "`name` must only contain -, a-z, A-Z, or 0-9."
+  }
```

Comment:
Lets makes a minimum length required for name something like at least 2 chars some resouces so have a minimum of 5 chars.